packer/core.go

```diff
@@ -374,6 +377,85 @@ func (c *Core) Context() *interpolate.Context {
 	}
 }
 
+func (c *Core) FixConfig(opts FixConfigOptions) hcl.Diagnostics {
+	var diags hcl.Diagnostics
+
+	var rawTemplateData map[string]interface{}
```

Comment:
Currently this will check the whole template file for fixable configuration fields, which will also appear when using the `-onl`y or `-except` CLI flags. Should we try to filter fix checks to only the builds specified in `-only` or `-except` as well?

Reply:
Given that the validate command does say only and except are valid cli options, I think it is appropriate to filter. Though realistically I kind of doubt anyone is running "validate" against individual builds inside a template file.

Reply:
ah to add to that [after some slack discussion] theses options were always available there but they were simply ignored. So I think this could be another PR ( Like an user actually asks for it ). What do you think ?

Reply:
Thanks @azr 

@SwampDragons to close the circle here. I was going down the same line of thinking and found myself in a situation where I would need to refactor the config types a bit to get filtering working in the FixConfig method. I could be over thinking it. 

Thinking a little more about your comment "folks running validate on individual builds" and the fact that the fix command operates on the whole file. I'm inclined to suggest that we skip the fixer check if a user is running a selective validate. Then work towards fully supporting it, if needed, in a future PR as @azr suggests.